src/pages/api/auth/[[...action]].ts

```diff
@@ -1,75 +1,113 @@
+    // Otherwise Passport won't initialize the strategy
     await configSaml(passport)
     next()
   })
   .use(passport.initialize())
-  .get('/api/auth/login', passport.authenticate('saml'))
-  .post('/api/auth/login', (req, res) => {
-    passport.authenticate('saml', function (err, user) {
+  .use(passport.session())
+
+// GET /api/auth/user - check for logged in user
+handler.get('/api/auth/user', (req, res) => {
+  if (req.isAuthenticated()) {
+    res.json({ user: req.user })
+  } else {
+    res.status(401).json({ user: null })
+  }
+})
+
+// GET /api/auth/login - initiate a SAML request
+handler.get('/api/auth/login', passport.authenticate('saml'))
+
+// POST /api/auth/login - callback URL for IdP SSO
+handler.post('/api/auth/login', (req, res) => {
+  passport.authenticate('saml', function (err, user) {
+    if (err) {
+      // TODO - error handling
+      // eslint-disable-next-line no-console
+      console.error('Error authenticating with SAML', err)
+      res.status(401).end()
+    }
+
+    if (!user) {
+      // TODO - error handling
+      // eslint-disable-next-line no-console
+      console.error('Error - missing user in SAML response', err)
+      res.status(401).end()
+    }
+
+    req.login(user, (err) => {
       if (err) {
         // TODO - error handling
         // eslint-disable-next-line no-console
-        console.error('Error authenticating with SAML', err)
-        res.redirect('/login')
+        console.error('Error logging user into', err)
+        res.status(401).end()
       }
 
-      if (!user) {
-        // TODO - error handling
-        // eslint-disable-next-line no-console
-        console.error('Error missing user in SAML response', err)
-        res.redirect('/login')
-      }
+      res.status(200).json({ user })
+    })
+  })(req, res)
+})
 
-      // TODO - set user in session & redirect
-      SESSION_USER = user
-      res.send(
-        `Logged in as user: ${user.edipi} ${user.userprincipalname} ${user['iv-groups']}`
-      )
-    })(req, res)
-  })
-  .get('/api/auth/logout', (req, res) => {
-    // Modify the NextApiRequest to include user attributes required by SLO
-    const logoutRequest = req as LogoutRequest
-    logoutRequest.user = {
-      nameID: SESSION_USER.nameID,
-      nameIDFormat: SESSION_USER.nameIDFormat,
-    }
+//  GET /api/auth/logout - initiate a SLO (logout) request
+handler.get('/api/auth/logout', async (req, res) => {
+  if (!req.user) {
+    // Not logged in
+    await req.session.destroy()
+    res.status(200).end()
+  } else {
+    await req.session.destroy()
 
-    passport.logoutSaml(logoutRequest, (err, request) => {
-      if (!err && request) {
-        res.redirect(request)
+    passport.logoutSaml(req, async (err, logoutRequest) => {
+      if (!err && logoutRequest) {
+        // TEMPORARY - DEVELOPMENT ONLY
+        // because our test IDP does not support SLO HTTP-POST bindings
+        if (
```

Comment:
I added this because the test IDP docker image we are using does not seem to allow us to configure the logout endpoint to accept HTTP-POST requests, only HTTP-Redirect. However the actual GCDS integration _only_ supports HTTP-POST logout. Long term I found a node-based [test IDP](https://github.com/mcguinness/saml-idp) that might be worth configuring ourselves for development to more closely mirror our actual auth, but that is not a priority right now.